src/main/java/org/testng/internal/thread/graph/GraphThreadPoolExecutor.java

```diff
@@ -26,18 +27,16 @@
   /** Set to true if you want to generate GraphViz graphs */
   private static final boolean DOT_FILES = false;
 
-  private DynamicGraph<T> m_graph;
+  private final DynamicGraph<T> m_graph;
   private List<Runnable> m_activeRunnables = Collections.synchronizedList(Lists.<Runnable>newArrayList());
   private IThreadWorkerFactory<T> m_factory;
   private List<String> m_dotFiles = Lists.newArrayList();
-  private int m_threadCount;
+  private boolean useLatches;
 
   public GraphThreadPoolExecutor(DynamicGraph<T> graph, IThreadWorkerFactory<T> factory, int corePoolSize,
       int maximumPoolSize, long keepAliveTime, TimeUnit unit, BlockingQueue<Runnable> workQueue) {
-    super(corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue
-        /* , new TestNGThreadPoolFactory() */);
```

Comment:
@juherr - I have rebased off of your changes that @cbeust  merged to master. So this is also taken care of.